src-backend/api/tests/test_generator.py

```diff
@@ -0,0 +1,274 @@
+        assert_true('id' in self.attribs)
+        assert_equals(self.attribs['id'], self.element.eid)
+
+    def test_element_has_type(self):
+        assert_true('type' in self.attribs)
+        assert_equals(self.attribs['type'], self.element.element_type)
+
+    def test_element_has_concept(self):
+        assert_true('concept' in self.attribs)
+        assert_equals(self.attribs['concept'], self.element.concept)
+
+    def test_element_has_question(self):
+        assert_true('question' in self.attribs)
+        assert_equals(self.attribs['question'], self.element.question)
+
+    def test_element_has_answer(self):
+        assert_true('answer' in self.attribs)
+        assert_equals(self.attribs['answer'], self.element.answer)
+
+    def test_element_has_no_numeric(self):
+        assert_false('numeric' in self.attribs)
+
+    def test_element_has_no_choices(self):
+        assert_false('choices' in self.attribs)
+
+    def test_element_has_numeric(self):
+        self.element.numeric = 'DIALPAD'
+        self.element_etree_element = self.generator.generate(ElementTree.Element('test'))
+
+        print self.element_etree_element.attrib
+
+        assert_true('numeric' in self.element_etree_element.attrib)
+        assert_equals(self.element_etree_element.attrib['numeric'], self.element.numeric)
+
+    def test_element_has_choices(self):
+        self.element.choices = 'left,right,center'
+        self.element_etree_element = self.generator.generate(ElementTree.Element('test'))
+
+        assert_true('choices' in self.element_etree_element.attrib)
+        assert_equals(self.element_etree_element.attrib['choices'], self.element.choices)
+
+
+class ProtocolBuilderTestCase(TestCase):
+    def setUp(self):
+        self.test_user = User.objects.create_user(
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
+        self.test_user.save()
+
+        self.procedure = Procedure.objects.create(
+            author='TestUser',
+            title='Burns',
+            owner=self.test_user
+        )
+
+        page1 = Page.objects.create(
+            display_index=0,
+            procedure=self.procedure
+        )
+        page2 = Page.objects.create(
+            display_index=1,
+            procedure=self.procedure
+        )
+        page3 = Page.objects.create(
+            display_index=2,
+            procedure=self.procedure
+        )
+        page4 = Page.objects.create(
+            display_index=3,
+            procedure=self.procedure
+        )
+
+        Element.objects.create(
+            display_index=0,
+            eid='burns',
+            element_type='MULTI_SELECT',
+            concept='BURNS',
+            question='Select One or More of the Following:',
+            answer='',
+            choices='On the head,On the chest,Loins,Inhalation burns,Chemical,In an enclosure,Patient Unconscious,Epileptic Patient, Diabetic Patient',  # noqa
```

Comment:
they should (or *need*) to be stored in a format that can serialize an arbitrary string array (i.e. comma delimited probably wont' be enough)